### PR TITLE
Python: fix +tkinter+tix support

### LINKED ISSUE #23780: Run-time issue: python+tkinter
The tkagg backend does not work as expected, see below with a fresh Spack at 319f20d94ff68f781b4913d236e0587dfa0195ba.

```console
$ spack install py-matplotlib backend=tkagg
$ spack load -r py-matplotlib
$ python -c "import matplotlib.pyplot as plt; plt.plot([0],[1]); plt.show()"
...
_tkinter.TclError: Can't find a usable tk.tcl in the following directories: 
    /home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/tk-8.6.10-egp4nzel66d2273p3olurpjif6bhhkmj/lib /home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/tcl-8.6.11-uoh3u6zmjwbky6rhfojdihz6blnk5sy5/lib/tcl8.6/tk8.6 /home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/tcl-8.6.11-uoh3u6zmjwbky6rhfojdihz6blnk5sy5/lib/tk8.6 /home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/python-3.8.10-efzidqekqhq4y4raa2q4iabetuoqdnse/lib/tk8.6 /home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/python-3.8.10-efzidqekqhq4y4raa2q4iabetuoqdnse/lib/tk8.6 /home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/lib/tk8.6 /home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/python-3.8.10-efzidqekqhq4y4raa2q4iabetuoqdnse/library
```

The variants `tk` and `tcl` itself were installed:

```console
$ spack find --loaded -v | grep -e tcl -e tk   
py-matplotlib@3.4.2~animation~fonts+image~latex~movies backend=tkagg
python@3.8.10+bz2+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tix+tkinter~ucs4+uuid+zlib patches=0d98e93189bc278fbc37a50ed7f183bd8aaf249a8e1670a465f0db6bb4f8cf87
tcl@8.6.11
tk@8.6.10+xft+xss
``` 

### General information

* **Spack:** 0.16.1-2700-319f20d94f
* **Python:** 3.8.5
* **Platform:** linux-ubuntu20.04-haswell
* **Concretizer:** original

### PR DESCRIPTION
**Pull request by adamjstewart**
Previously, we were setting `TCLLIBPATH` to `prefix/lib/tcl8.6`. With this PR, we are now setting `TCLLIBPATH` to `prefix/lib`. As far as I can tell from reading https://wiki.tcl-lang.org/page/TCLLIBPATH, this is the correct way to do things.

We are also now setting `(TCL|TK|TIX)_LIBRARY` correctly. Tix now builds for me.

Fixes #23780. The following commands successfully run:
```console
$ spack install python+tkinter
$ spack load python+tkinter
$ python -c 'import tkinter; tkinter._test()'
```

@skosukhin you can check Python if you want to but I think I have it thoroughly tested.

@glennpj can you make sure this doesn't break R?

We DESPERATELY need an official maintainer for the Tcl/Tk packages, as evidenced by the dozens of comments in the package from lessons hard learned. Pinging some people who have contributed to our Tcl/Tk recipes in the past:
@nazavode @zzzoom @xdelaruelle @gardner48 @lee218llnl @sknigh @citibeth @erimar77 @mjwoods @michaelkuhn 

Also see:

* https://core.tcl-lang.org/tk/tktview/447bd3e4abe17452d19a80e6840dcc8a2603fcbc
* https://bugs.python.org/issue44253

### CODE REVIEW

**Review by alalazo — DISMISSED** [review]

**Comment by schmitts** [comment]
@adamjstewart unfortunately it stills fails

```
_tkinter.TclError: Can't find a usable tk.tcl in the following directories:

/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/tk-8.6.11-3pygupoqtd3fixjimjfyu34o6zesxrgu/lib
/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/tcl-8.6.11-jgl7fpo5wz37kcve7pskehab2tbnr4h7/lib/tk8.6
/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/tcl-8.6.11-jgl7fpo5wz37kcve7pskehab2tbnr4h7/lib/tcl8.6/tk8.6
/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/python-3.8.10-y5pziqe7qun4n6ks5g42lr2eh6erkldl/lib/tk8.6
/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/python-3.8.10-y5pziqe7qun4n6ks5g42lr2eh6erkldl/lib/tk8.6
/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/lib/tk8.6
/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/python-3.8.10-y5pziqe7qun4n6ks5g42lr2eh6erkldl/library

sschmitt@sierra ➜  ~/spack-tcl git:(packages/tcl) find -name tk.tcl
./opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/tk-8.6.11-3pygupoqtd3fixjimjfyu34o6zesxrgu/share/tk/src/library/tk.tcl
./opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/tk-8.6.11-3pygupoqtd3fixjimjfyu34o6zesxrgu/lib/tk8.6/tk.tcl
```

**Comment by adamjstewart** [comment]
@schmitts can you echo $TCLLIBPATH?

**Comment by schmitts** [comment]
> @schmitts can you echo $TCLLIBPATH?

```console
echo $TCLLIBPATH
/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/tcl-8.6.11-jgl7fpo5wz37kcve7pskehab2tbnr4h7/lib
```

**Comment by adamjstewart** [comment]
If you add tk's lib dir to the path, does that help?

**Comment by adamjstewart** [comment]
After running `spack load python+tkinter`, my env looks like:
```console
$ env | egrep 'TCL|TK'
TK_LIBRARY=/Users/Adam/spack/opt/spack/darwin-catalina-x86_64/apple-clang-12.0.0/tk-8.6.11-ydmhrbboheucxsuhrnyoxqaihgna5dfe/lib
TCLLIBPATH=/Users/Adam/spack/opt/spack/darwin-catalina-x86_64/apple-clang-12.0.0/tcl-8.6.11-n7nea33urrk25rkoqpsc2tdcgai5u4z2/lib /Users/Adam/spack/opt/spack/darwin-catalina-x86_64/apple-clang-12.0.0/tk-8.6.11-ydmhrbboheucxsuhrnyoxqaihgna5dfe/lib
TCL_LIBRARY=/Users/Adam/spack/opt/spack/darwin-catalina-x86_64/apple-clang-12.0.0/tcl-8.6.11-n7nea33urrk25rkoqpsc2tdcgai5u4z2/lib
```
Running the tkinter tests opens up a window I can click on and quit from.

**Comment by adamjstewart** [comment]
Hmm, even for me this seems to work sometimes and not other times. Been playing around with different combinations of env vars but nothing seems to work reliably. If I have the above env vars set, I don't see the error message, but things like `python -m tkinter` and `python -c 'import tkinter; tkinter._test()` will open a window and immediately minimize it for some reason.

**Comment by adamjstewart** [comment]
This is also useful for debugging: https://stackoverflow.com/questions/43464355/how-to-find-the-path-of-tcl-tk-library-that-tkinter-is-currently-using/43464686

**Comment by glennpj** [comment]
@adamjstewart Do you still need me to verify R or is there more to do for this?

**Comment by adamjstewart** [comment]
Honestly, not sure. I asked the Python and Tcl developers to comment on which env vars are used and what they should be set to. Hoping to get a response from them before doing any more hacking.

**Comment by adamjstewart** [comment]
According to https://bugs.python.org/issue31817, `TK_LIBRARY` is used and we're setting it to the wrong location. I wish I could get a straight answer as to what env vars are used by what and what they need to be set to...

**Comment by adamjstewart** [comment]
https://bugs.python.org/issue20035 also contains an interesting discussion of `(TCL|TK|TIX)_LIBRARY`

**Comment by adamjstewart** [comment]
@schmitts can you try again with the latest commit? I'm now seeing:
```console
$ spack install python+tkinter+tix
$ spack load python+tkinter+tix
$ env | egrep 'TCL|TK|TIX'
TK_LIBRARY=/Users/Adam/spack/opt/spack/darwin-catalina-x86_64/apple-clang-12.0.0/tk-8.6.11-ydmhrbboheucxsuhrnyoxqaihgna5dfe/lib/tk8.6
TIX_LIBRARY=/Users/Adam/spack/opt/spack/darwin-catalina-x86_64/apple-clang-12.0.0/tix-8.4.3-7uo5u3c23kcmj43rysm3paw3iyfvknuz/lib/Tix8.4.3
TCLLIBPATH=/Users/Adam/spack/opt/spack/darwin-catalina-x86_64/apple-clang-12.0.0/tix-8.4.3-7uo5u3c23kcmj43rysm3paw3iyfvknuz/lib /Users/Adam/spack/opt/spack/darwin-catalina-x86_64/apple-clang-12.0.0/tk-8.6.11-ydmhrbboheucxsuhrnyoxqaihgna5dfe/lib
TCL_LIBRARY=/Users/Adam/spack/opt/spack/darwin-catalina-x86_64/apple-clang-12.0.0/tcl-8.6.11-n7nea33urrk25rkoqpsc2tdcgai5u4z2/lib/tcl8.6
$ tclsh
% puts $auto_path
/Users/Adam/spack/opt/spack/darwin-catalina-x86_64/apple-clang-12.0.0/tix-8.4.3-7uo5u3c23kcmj43rysm3paw3iyfvknuz/lib /Users/Adam/spack/opt/spack/darwin-catalina-x86_64/apple-clang-12.0.0/tk-8.6.11-ydmhrbboheucxsuhrnyoxqaihgna5dfe/lib /Users/Adam/spack/opt/spack/darwin-catalina-x86_64/apple-clang-12.0.0/tcl-8.6.11-n7nea33urrk25rkoqpsc2tdcgai5u4z2/lib /Users/Adam/spack/opt/spack/darwin-catalina-x86_64/apple-clang-12.0.0/tcl-8.6.11-n7nea33urrk25rkoqpsc2tdcgai5u4z2
$ python
Python 3.8.10 (default, May 29 2021, 17:39:35) 
[Clang 12.0.0 (clang-1200.0.32.29)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import tkinter
>>> root = tkinter.Tk()
>>> from tkinter import tix
>>> root = tix.Tk()
>>> print(root.tk.exprstring('$tcl_library'))
/Users/Adam/spack/opt/spack/darwin-catalina-x86_64/apple-clang-12.0.0/tcl-8.6.11-n7nea33urrk25rkoqpsc2tdcgai5u4z2/lib
>>> print(root.tk.exprstring('$tk_library'))
/Users/Adam/spack/opt/spack/darwin-catalina-x86_64/apple-clang-12.0.0/tk-8.6.11-ydmhrbboheucxsuhrnyoxqaihgna5dfe/lib/tk8.6
>>> print(root.tk.exprstring('$tix_library'))
/Users/Adam/spack/opt/spack/darwin-catalina-x86_64/apple-clang-12.0.0/tix-8.4.3-7uo5u3c23kcmj43rysm3paw3iyfvknuz/lib/Tix8.4.3
>>> tkinter._test()
```
I've come to the conclusion that `TCLLIBPATH` isn't actually important at all and `(TCL|TK|TIX)_LIBRARY` determines everything, at least for Python.

**Comment by schmitts** [comment]
Alas, a new error:

```console
sschmitt@sierra ➜  ~/spack-tcl git:(packages/tcl) spack load -r py-matplotlib backend=tkagg
sschmitt@sierra ➜  ~/spack-tcl git:(packages/tcl) env | egrep 'TCL|TK'                     
GTK2_RC_FILES=/etc/gtk-2.0/gtkrc:/home/sschmitt/.gtkrc-2.0:/home/sschmitt/.config/gtkrc-2.0
GTK_RC_FILES=/etc/gtk/gtkrc:/home/sschmitt/.gtkrc:/home/sschmitt/.config/gtkrc
TK_LIBRARY=/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/tk-8.6.11-3pygupoqtd3fixjimjfyu34o6zesxrgu/lib/tk8.6
TCLLIBPATH=/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/tk-8.6.11-3pygupoqtd3fixjimjfyu34o6zesxrgu/lib
TCL_LIBRARY=/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/tcl-8.6.11-jgl7fpo5wz37kcve7pskehab2tbnr4h7/lib/tcl8.6
sschmitt@sierra ➜  ~/spack-tcl git:(packages/tcl) python -c 'import tkinter; tkinter._test()'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/python-3.8.10-y5pziqe7qun4n6ks5g42lr2eh6erkldl/lib/python3.8/tkinter/__init__.py", line 4557, in _test
    root = Tk()
  File "/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/python-3.8.10-y5pziqe7qun4n6ks5g42lr2eh6erkldl/lib/python3.8/tkinter/__init__.py", line 2270, in __init__
    self.tk = _tkinter.create(screenName, baseName, className, interactive, wantobjects, useTk, sync, use)
_tkinter.TclError: invalid command name "tcl_findLibrary"
```

**Comment by adamjstewart** [comment]
@schmitts try again now, I think I finally have it worked out. You should see:
```console
$ spack install python+tkinter+tix
$ spack load python+tkinter+tix
$ env | egrep 'TCL|TK|TIX'
TK_LIBRARY=/Users/Adam/spack/opt/spack/darwin-catalina-x86_64/apple-clang-12.0.0/tk-8.6.11-ydmhrbboheucxsuhrnyoxqaihgna5dfe/lib/tk8.6
TIX_LIBRARY=/Users/Adam/spack/opt/spack/darwin-catalina-x86_64/apple-clang-12.0.0/tix-8.4.3-7uo5u3c23kcmj43rysm3paw3iyfvknuz/lib/Tix8.4.3
TCLLIBPATH=/Users/Adam/spack/opt/spack/darwin-catalina-x86_64/apple-clang-12.0.0/tix-8.4.3-7uo5u3c23kcmj43rysm3paw3iyfvknuz/lib /Users/Adam/spack/opt/spack/darwin-catalina-x86_64/apple-clang-12.0.0/tk-8.6.11-ydmhrbboheucxsuhrnyoxqaihgna5dfe/lib
TCL_LIBRARY=/Users/Adam/spack/opt/spack/darwin-catalina-x86_64/apple-clang-12.0.0/tcl-8.6.11-n7nea33urrk25rkoqpsc2tdcgai5u4z2/lib
$ tclsh
% puts $auto_path
/Users/Adam/spack/opt/spack/darwin-catalina-x86_64/apple-clang-12.0.0/tix-8.4.3-7uo5u3c23kcmj43rysm3paw3iyfvknuz/lib /Users/Adam/spack/opt/spack/darwin-catalina-x86_64/apple-clang-12.0.0/tk-8.6.11-ydmhrbboheucxsuhrnyoxqaihgna5dfe/lib /Users/Adam/spack/opt/spack/darwin-catalina-x86_64/apple-clang-12.0.0/tcl-8.6.11-n7nea33urrk25rkoqpsc2tdcgai5u4z2/lib /Users/Adam/spack/opt/spack/darwin-catalina-x86_64/apple-clang-12.0.0/tcl-8.6.11-n7nea33urrk25rkoqpsc2tdcgai5u4z2
$ python
Python 3.8.10 (default, May 29 2021, 17:39:35) 
[Clang 12.0.0 (clang-1200.0.32.29)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import tkinter
>>> root = tkinter.Tk()
>>> from tkinter import tix
>>> root = tix.Tk()
>>> print(root.tk.exprstring('$tcl_library'))
/Users/Adam/spack/opt/spack/darwin-catalina-x86_64/apple-clang-12.0.0/tcl-8.6.11-n7nea33urrk25rkoqpsc2tdcgai5u4z2/lib
>>> print(root.tk.exprstring('$tk_library'))
/Users/Adam/spack/opt/spack/darwin-catalina-x86_64/apple-clang-12.0.0/tk-8.6.11-ydmhrbboheucxsuhrnyoxqaihgna5dfe/lib/tk8.6
>>> print(root.tk.exprstring('$tix_library'))
/Users/Adam/spack/opt/spack/darwin-catalina-x86_64/apple-clang-12.0.0/tix-8.4.3-7uo5u3c23kcmj43rysm3paw3iyfvknuz/lib/Tix8.4.3
>>> tkinter._test()
```
Your matplotlib plotting example also works for me.

**Comment by schmitts** [comment]
I can confirm that

```console
python -c 'import tkinter; tkinter._test()'
```

and

```console
python -c "import matplotlib.pyplot as plt; plt.plot([0],[1]); plt.show()"
```

works now as expected.

```console
TCLLIBPATH=/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/tk-8.6.11-3pygupoqtd3fixjimjfyu34o6zesxrgu/lib
TCL_LIBRARY=/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/tcl-8.6.11-jgl7fpo5wz37kcve7pskehab2tbnr4h7/lib/tcl8.6
TK_LIBRARY=/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/tk-8.6.11-3pygupoqtd3fixjimjfyu34o6zesxrgu/lib/tk8.6
```

```console
>>> import tkinter
>>> root = tkinter.Tk()
>>> print(root.tk.exprstring('$tcl_library'))
/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/tcl-8.6.11-jgl7fpo5wz37kcve7pskehab2tbnr4h7/lib/tcl8.6
>>> print(root.tk.exprstring('$tcl_library'))
/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/tcl-8.6.11-jgl7fpo5wz37kcve7pskehab2tbnr4h7/lib/tcl8.6
```

**Comment by schmitts** [comment]
[spack-build-out.txt.gz](https://github.com/spack/spack/files/6568596/spack-build-out.txt.gz)
But installing `tix` fails:

```console
==> Installing tix-8.4.3-r7y6mmznctw7pskk2xkfgnpfg7fffxdz
==> No binary for tix-8.4.3-r7y6mmznctw7pskk2xkfgnpfg7fffxdz found: installing from source
==> Fetching https://spack-llnl-mirror.s3-us-west-2.amazonaws.com/_source-cache/archive/56/562f040ff7657e10b5cffc2c41935f1a53c6402eb3d5f3189113d734fd6c03cb.tar.gz
#=#=-  #       #                                                                                                                                                                                                                                               #=O=#     #        #                                                                                                                                                                                                                                           -#O=- #      #          #       ######################################################################################################################################################################################################################################################### 100.0%
==> Fetching https://raw.githubusercontent.com/macports/macports-ports/master/x11/tix/files/panic.patch
######################################################################################################################################################################################################################################################### 100.0%
==> Fetching https://raw.githubusercontent.com/macports/macports-ports/master/x11/tix/files/patch-generic-tixGrSort.c.diff
######################################################################################################################################################################################################################################################### 100.0%
==> Fetching https://raw.githubusercontent.com/macports/macports-ports/master/x11/tix/files/implicit.patch
######################################################################################################################################################################################################################################################### 100.0%
==> Applied patch https://raw.githubusercontent.com/macports/macports-ports/master/x11/tix/files/panic.patch
==> Applied patch https://raw.githubusercontent.com/macports/macports-ports/master/x11/tix/files/patch-generic-tixGrSort.c.diff
==> Applied patch https://raw.githubusercontent.com/macports/macports-ports/master/x11/tix/files/implicit.patch
==> tix: Executing phase: 'autoreconf'
==> tix: Executing phase: 'configure'
==> tix: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j8'

8 errors found in build log:
     94     /home/sschmitt/spack-tcl/lib/spack/env/gcc/gcc -DPACKAGE_NAME=\"Tix\" -DPACKAGE_TARNAME=\"tix\" -DPACKAGE_VERSION=\"8.4.3\" -DPACKAGE_STRING=\"Tix\ 8.4.3\" -DPACKAGE_BUGREPORT=\"\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -
            DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_LIMITS_H=1 -DHAVE_SYS_PARAM_H=1 -DUSE_THREAD_ALLOC=1 -D_REENTRANT=1 -D_THREAD_SAFE=1 -DTCL_THREADS=1 -D_LARGEFILE64_SOURCE=1 -DTCL_WIDE_INT_I
            S_LONG=1 -DUSE_TCL_STUBS=1 -DUSE_TK_STUBS=1   -I. -I"/tmp/sschmitt/spack-stage/spack-stage-tix-8.4.3-r7y6mmznctw7pskk2xkfgnpfg7fffxdz/spack-src/generic" -I. -I"/tmp/sschmitt/spack-stage/spack-stage-tix-8.4.3-r7y6mmznctw7pskk2xkfgnpfg7fffxdz/spack-src/unix" -
            I"/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/tcl-8.6.11-jgl7fpo5wz37kcve7pskehab2tbnr4h7/share/tcl/src/generic" -I"/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/tcl-8.6.11-jgl7fpo5wz37kcve7pskehab2tbnr4h7/
            share/tcl/src/unix" -I"/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/tk-8.6.11-3pygupoqtd3fixjimjfyu34o6zesxrgu/share/tk/src/generic" -I"/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/tk-8.6.11-3pygupoqtd3fixj
            imjfyu34o6zesxrgu/share/tk/src/unix" -I/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/libx11-1.7.0-3ukzesrsqcspevi4a6svf3ynxalrwpx7/include    -pipe -O2 -fomit-frame-pointer -Wall -Wno-implicit-int -fPIC  -c `echo /tmp/sschmitt/spack-
            stage/spack-stage-tix-8.4.3-r7y6mmznctw7pskk2xkfgnpfg7fffxdz/spack-src/generic/tixInit.c`
     95     /home/sschmitt/spack-tcl/lib/spack/env/gcc/gcc -DPACKAGE_NAME=\"Tix\" -DPACKAGE_TARNAME=\"tix\" -DPACKAGE_VERSION=\"8.4.3\" -DPACKAGE_STRING=\"Tix\ 8.4.3\" -DPACKAGE_BUGREPORT=\"\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -
            DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_LIMITS_H=1 -DHAVE_SYS_PARAM_H=1 -DUSE_THREAD_ALLOC=1 -D_REENTRANT=1 -D_THREAD_SAFE=1 -DTCL_THREADS=1 -D_LARGEFILE64_SOURCE=1 -DTCL_WIDE_INT_I
            S_LONG=1 -DUSE_TCL_STUBS=1 -DUSE_TK_STUBS=1   -I. -I"/tmp/sschmitt/spack-stage/spack-stage-tix-8.4.3-r7y6mmznctw7pskk2xkfgnpfg7fffxdz/spack-src/generic" -I. -I"/tmp/sschmitt/spack-stage/spack-stage-tix-8.4.3-r7y6mmznctw7pskk2xkfgnpfg7fffxdz/spack-src/unix" -
            I"/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/tcl-8.6.11-jgl7fpo5wz37kcve7pskehab2tbnr4h7/share/tcl/src/generic" -I"/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/tcl-8.6.11-jgl7fpo5wz37kcve7pskehab2tbnr4h7/
            share/tcl/src/unix" -I"/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/tk-8.6.11-3pygupoqtd3fixjimjfyu34o6zesxrgu/share/tk/src/generic" -I"/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/tk-8.6.11-3pygupoqtd3fixj
            imjfyu34o6zesxrgu/share/tk/src/unix" -I/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/libx11-1.7.0-3ukzesrsqcspevi4a6svf3ynxalrwpx7/include    -pipe -O2 -fomit-frame-pointer -Wall -Wno-implicit-int -fPIC  -c `echo /tmp/sschmitt/spack-
            stage/spack-stage-tix-8.4.3-r7y6mmznctw7pskk2xkfgnpfg7fffxdz/spack-src/generic/tixList.c`
     96     /home/sschmitt/spack-tcl/lib/spack/env/gcc/gcc -DPACKAGE_NAME=\"Tix\" -DPACKAGE_TARNAME=\"tix\" -DPACKAGE_VERSION=\"8.4.3\" -DPACKAGE_STRING=\"Tix\ 8.4.3\" -DPACKAGE_BUGREPORT=\"\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -
            DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_LIMITS_H=1 -DHAVE_SYS_PARAM_H=1 -DUSE_THREAD_ALLOC=1 -D_REENTRANT=1 -D_THREAD_SAFE=1 -DTCL_THREADS=1 -D_LARGEFILE64_SOURCE=1 -DTCL_WIDE_INT_I
            S_LONG=1 -DUSE_TCL_STUBS=1 -DUSE_TK_STUBS=1   -I. -I"/tmp/sschmitt/spack-stage/spack-stage-tix-8.4.3-r7y6mmznctw7pskk2xkfgnpfg7fffxdz/spack-src/generic" -I. -I"/tmp/sschmitt/spack-stage/spack-stage-tix-8.4.3-r7y6mmznctw7pskk2xkfgnpfg7fffxdz/spack-src/unix" -
            I"/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/tcl-8.6.11-jgl7fpo5wz37kcve7pskehab2tbnr4h7/share/tcl/src/generic" -I"/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/tcl-8.6.11-jgl7fpo5wz37kcve7pskehab2tbnr4h7/
            share/tcl/src/unix" -I"/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/tk-8.6.11-3pygupoqtd3fixjimjfyu34o6zesxrgu/share/tk/src/generic" -I"/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/tk-8.6.11-3pygupoqtd3fixj
            imjfyu34o6zesxrgu/share/tk/src/unix" -I/home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/libx11-1.7.0-3ukzesrsqcspevi4a6svf3ynxalrwpx7/include    -pipe -O2 -fomit-frame-pointer -Wall -Wno-implicit-int -fPIC  -c `echo /tmp/sschmitt/spack-
            stage/spack-stage-tix-8.4.3-r7y6mmznctw7pskk2xkfgnpfg7fffxdz/spack-src/generic/tixMethod.c`
     97     In file included from /home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/tcl-8.6.11-jgl7fpo5wz37kcve7pskehab2tbnr4h7/share/tcl/src/generic/tcl.h:2441,
     98                      from /tmp/sschmitt/spack-stage/spack-stage-tix-8.4.3-r7y6mmznctw7pskk2xkfgnpfg7fffxdz/spack-src/generic/tixPort.h:22,
     99                      from /tmp/sschmitt/spack-stage/spack-stage-tix-8.4.3-r7y6mmznctw7pskk2xkfgnpfg7fffxdz/spack-src/generic/tixGeometry.c:16:
  >> 100    /home/sschmitt/spack-tcl/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/tcl-8.6.11-jgl7fpo5wz37kcve7pskehab2tbnr4h7/share/tcl/src/generic/tclDecls.h:2542:14: error: expected ')' before '->' token
     101     2542 |  (tclStubsPtr->tcl_Panic) /* 2 */
```

**Comment by adamjstewart** [comment]
@schmitts I'm able to reproduce this build error on CentOS 7 with GCC 4.8.5. I tried adding all of the patches from https://github.com/macports/macports-ports/tree/master/x11/tix/files but unfortunately none of them solved it. I don't know C well enough to fix it myself, and I'm not sure where else to look for patches. https://sourceforge.net/p/tix/patches/ doesn't seem to have anything for this specific build error.

Basically, I think this PR is ready to go, as it fixes `python+tkinter` and fixes the `tix` build on macOS, and doesn't make the build any worse on Linux. But if you can find/figure out a patch to fix the build on Linux then I'll be happy to add it.

**Comment by adamjstewart** [comment]
@glennpj can you test this with R now?

**Comment by glennpj** [comment]
@adamjstewart I should be able to test R tomorrow, sorry for the delay.

**Comment by glennpj** [comment]
@adamjstewart This PR works fine with R. I built fresh `tcl` and `tk` packages along with a fresh build of `r` and `r-misc3d`, which explicitly looks for `tcltk` capability.

```
Rscript -e 'capabilities("tcltk")'
tcltk 
 TRUE 
```
The misc3d library loads without any issues.

**Comment by adamjstewart** [comment]
@alalazo I think this is ready for a final review. I'm pretty confident about this now that the Tcl developers confirmed what `TCLLIBPATH` should look like and we have smoke tests now.

**Review by alalazo — APPROVED** [review]
